crates/re_viewport/src/context_menu/mod.rs

```diff
@@ -69,232 +77,219 @@ pub fn context_menu_ui_for_item(
                         .set_selection(std::iter::once(item.clone()));
                 }
 
-                summarize_selection(&Selection::from(item.clone()))
+                show_context_menu(&Selection::from(item.clone()));
             }
 
-            SelectionUpdateBehavior::Ignore => summarize_selection(&Selection::from(item.clone())),
+            SelectionUpdateBehavior::Ignore => show_context_menu(&Selection::from(item.clone())),
         };
+    });
+}
 
-        let actions = context_menu_items_for_selection_summary(
-            ctx,
-            viewport_blueprint,
-            item,
-            selection_summary,
-        );
-
-        if actions.is_empty() {
-            ui.label(
-                egui::RichText::from("No action available for the current selection").italics(),
-            );
-        }
+/// Returns the (statically-defined) list of action, grouped in sections.
+///
+/// Sections are group of actions that should be displayed together, with a separator displayed
+/// between sections.
+fn action_list(
+    ctx: &ViewerContext<'_>,
+) -> &'static Vec<Vec<Box<dyn ContextMenuAction + Sync + Send>>> {
+    static CONTEXT_MENU_ACTIONS: OnceCell<Vec<Vec<Box<dyn ContextMenuAction + Sync + Send>>>> =
+        OnceCell::new();
+
+    CONTEXT_MENU_ACTIONS.get_or_init(|| {
```

Comment:
this makes me itch to write another registry :D. Well, later maybe.

Reply:
Not to worry, we'll probably end up getting there :)